lib/jekyll-admin/directory.rb

```diff
@@ -0,0 +1,58 @@
+module JekyllAdmin
+  class Directory
+    extend Forwardable
+    def_delegator :@path, :basename, :name
+    def_delegator :@path, :mtime, :modified_time
+    attr_reader :path, :splat, :content_type, :base
+
+    include Enumerable
+    include JekyllAdmin::URLable
+    include JekyllAdmin::APIable
+
+    TYPE = "directory".freeze
+
+    def initialize(path, base: nil, content_type: nil, splat: nil)
+      @base = Pathname.new base
+      @content_type = content_type
+      @splat = Pathname.new splat
+      @path = Pathname.new path
+    end
+
+    def to_liquid
+      {
+        :name          => name,
+        :modified_time => modified_time,
+        :path          => relative_path,
+        :type          => TYPE,
+      }
+    end
+
+    def relative_path
+      path.relative_path_from(base).to_s
+    end
+
+    def resource_path
+      if content_type == "pages"
```

Comment:
If it's coming from the URL, the this is correct as is. You never want to call `to_sym` on user input, as it's a DDoS vector.